core/chains/evm/logpoller/log_poller.go

```diff
@@ -346,7 +357,33 @@ func (lp *logPoller) run() {
 			} else {
 				start = lastProcessed.BlockNumber + 1
 			}
-			lp.pollAndSaveLogs(lp.ctx, start)
+			lp.pollAndSavePendingLogs(lp.ctx, start)
+		case <-logPollFinalizedTick:
+			// always run backup poller if it's been at least 10 mins since last run
```

Comment:
I think we can simplify - instead of saving both firstUnfinalizedBlockNumber and lastBackupPollerRun, we can just save backupPollLastBlock as the sole in memory state. Then we poll every 10 minutes and in each poll:
```
lastFinalizedBlock = (latest-lp.finalityDepth)
if lastFinalizedBlock > lp.backupPollLastBlock:
   backfill(backupPollLastBlock+1, lastFinalizedBlock)
   lp.backupPollLastBlock = lastFinalizedBlock 
```
That is to say, we just backup poll whatever new finalized blocks have been created in the last 10 minutes (never missing a finalized block).